### DISCUSSION TITLE
[tar]`@types/tar@6.1.3` Missing Event Names

### DISCUSSION THREAD

**Discussion by joealden** [DefinitelyTyped/DefinitelyTyped#63213] [Issues with a @types package]
We have some code in our codebase that looks something like this (simplified to highlight the problem):

```js
import { createGunzip } from "node:zlib";
import fetch from "node-fetch";
import tar from "tar";

const response = await fetch(...);
const stream = response.body.pipe(createGunzip({})).pipe(tar.t());

...

await new Promise<void>((resolve, reject) => {
  stream.on("finish", () => resolve());
  stream.on("error", error => reject(error));
});
```

With `@types/tar@6.1.2`, the two lines that start with `stream.on` are typed correctly, however with `@types/tar@6.1.3`, there are type errors on both lines regarding the event name passed. With `@types/tar@6.1.2`, the auto-complete options for the event name are `"close" | "drain" | "error" | "finish" | "pipe" | "unpipe"`, but with  `@types/tar@6.1.3`, the auto-complete options for the event name are `"close" | "end" | "entry"`.

[a comment by an automated account is not shown]

**Comment by joealden** [comment]
@SomaticIT @connor4312 I've had a little time to look into why this has happened, and it's because https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61949 change the return type of `list` when passing no args from `stream.Writable` to `Parse` (and `Parse` doesn't contain all events `stream.Writable` does). I don't have the knowledge to know why this might have been done, so I'm not sure what would need changing to restore the ability to listen for the other supported events.

cc @Nokel81 as you were the one that wrote the PR mentioned above!

**Reply by Nokel81** [reply]
It was changed because that is what the code does. If `Parse` is incomplete then it should be fixed.
